dozer/cogs/voice.py

```diff
@@ -27,6 +42,60 @@ async def on_voice_state_update(self, member, before, after):
                 if len(config) != 0:
                     await member.add_roles(member.guild.get_role(config[0].role_id))
 
+    @Cog.listener('on_voice_state_update')  # Used for auto PTT
+    async def on_PTT_check(self, member, before, after):
+        """Runs the autoPTTcheck when a user leaves and/or joins a vc"""
+        # skip this if we have no perms to edit voice channel
+        total_users = 0
+        if member.guild.me.guild_permissions.manage_channels and before.channel != after.channel:
+            # determine if it's a join/leave event as well.
+            # before and after are voice states
+            if before.channel is not None:
+                # leave event, take role
+                await self.auto_ptt_check(before)
+            if after.channel is not None:
+                # join event, give role
+                await self.auto_ptt_check(after)
+
+    @command()
+    @bot_has_permissions(manage_channels=True)
+    @has_permissions(manage_channels=True)
+    async def autoptt(self, ctx, voice_channel: discord.VoiceChannel, ptt_threshold: int):
+        """Configures AutoPtt limit for when members join/leave voice channels ptt is enabled"""
+
+        e = discord.Embed(color=blurple)
+        e.set_footer(text='Triggered by ' + ctx.author.display_name)
+
+        if ptt_threshold < 0:
+            raise BadArgument('PTT threshold must be positive integer')
+
+        if ptt_threshold == 0:
+            config = await AutoPTT.get_by(channel_id=voice_channel.id)
+            if len(config) != 0:
+                await AutoPTT.delete(id=config[0].id)
+                e.add_field(name='Success!', value='AutoPTT has been disabled for voice channel "**{}**"'
+                            .format(voice_channel))
+            else:
+                e.add_field(name='Error', value='AutoPTT has not been configured for voice channel "**{}**"'
+                            .format(voice_channel))
+        else:
+            ent = AutoPTT(
+                channel_id=voice_channel.id,
+                ptt_limit=ptt_threshold
+            )
+
+            await ent.update_or_add()
```

Comment:
With an autoincrementing primary key, this will always insert a new record instead of updating an existing one. Since the order records are returned from a `SELECT` statement isn't specified unless there's an `ORDER BY` clause, `auto_ptt_check` could fetch any record from the channel's history, which I'm pretty confident isn't the intent. If `channel_id` is the primary key for the `AutoPTT` table, this will work as expected.